lesson01/assignment/scripts.js

```diff
@@ -2,32 +2,143 @@
   })
 
-})
\ No newline at end of file
+  addingForm.addEventListener('submit', function(event){
+    // Prevent the form from trying to submit to a server
+    event.preventDefault();
+
+    // An array that we can transform based on user input
+    let data = [];
+
+    const inputField = document.querySelector('#adding-form .input');
+    const userInput = inputField.value;
+    const userNumber = parseInt(userInput);
+
+    // If the user's input was a number, push it into the array and continue
+    if(!isNaN(userNumber)){
+      data.push(userNumber);
+
+      // Use map to add 5 to the data in the array
+      const result = data.map(x => x + 5);
+
+      // Output the result to the page
+      document.querySelector('#adding-form .output').innerHTML = result;
+    }
+
+    // Clear the form field so the user can try again
+    inputField.value = '';
+  })
+
+  filterForm.addEventListener('submit', function(event){
+    // Prevent the form from trying to submit to a server
+    event.preventDefault();
+
+    // An array that we can transform based on user input
+    let data = [7, 10, 22, 5, 3];
+
+    const inputField = document.querySelector('#filter-form .input');
+    const userInput = inputField.value;
+    const userNumber = parseInt(userInput);
+
+    // If the user's input was a number, push it into the array and continue
+    if(!isNaN(userNumber)){
+      data.push(userNumber);
+
+      // Use filter to show all numbers larger than 7
+      const result = data.filter(x => x > 7);
+
+      // Output the result to the page
+      document.querySelector('#filter-form .output').innerHTML = result;
+    }
+
+    // Clear the form field so the user can try again
+    inputField.value = '';
+  })
+
+  includesForm.addEventListener('submit', function(event){
+    // Prevent the form from trying to submit to a server
+    event.preventDefault();
+
+    // An array that we can transform based on user input
+    let data = [];
+
+    const inputField = document.querySelector('#includes-form .input');
+    const userInput = inputField.value;
+    const userNumber = parseInt(userInput);
+
+    // If the user's input was a number, push it into the array and continue
+    if(!isNaN(userNumber)){
+      data.push(userNumber);
+
+      // Use includes to determine if there is a 7
+      const result = data.includes(7);
+
+      // Output the result to the page
+      document.querySelector('#includes-form .output').innerHTML = result;
+    }
+
+    // Clear the form field so the user can try again
+    inputField.value = '';
+  })
+
+  nameForm.addEventListener('submit', function(event){
+    // Prevent the form from trying to submit to a server
+    event.preventDefault();
+
+    // An array that we can transform based on user input
+    let data = ['Bob', 'Michael', 'Rachel', 'Zoe'];
+
+    const inputField = document.querySelector('#name-form .input');
+    const userInput = inputField.value;
+
+    if(isNaN(userInput)){
```

Comment:
No need to check that this isn't a number, all text form fields return a string.